### DISCUSSION TITLE
Plum juice

### DISCUSSION THREAD

**Discussion by unprompted** [paullouisageneau/libjuice#273] [Q&A] [answered]
Hi, first of all, libjuice is working well for my use case so far, so thanks for that @paullouisageneau!

Now I would like to do due diligence to limit unnecessary use of TURN for cost and latency reasons.  I'm under the impression that some subset of users will resort to TURN even though it might be avoided with some sort of PCP/NAT-PMP/UPnP.

So I was surprised to discover libplum today (I have some slight experience with miniupnpc) but haven't yet seen anything integrating both libjuice and libplum.

Since I'm already in over my head with acronyms I've only just learned, I wanted to ask if any of this even makes sense and if there is an obvious approach to integrating the two.  Will good things happen if I use libplum to forward the entire local_port_range before attempting ICE?  Is there any public code already using the two?  Or am I barking up the wrong tree entirely?

Thanks,
Cory

**Comment by paullouisageneau** [comment, marked as answer]
Thank you, I'm glad the library is useful to you!

This is good question. In theory, it could make sense for libjuice to leverage libplum. It wouldn't be done by port forwarding the whole range, instead each agent in libjuice would attempt to gather an additional host candidate by forwarding the existing ICE candidate port with libplum. However, this is outside of the ICE standard.

**Comment by unprompted** [comment]
Ah, yeah, that makes sense, and I think I see how I could achieve that if I decide I need it.  I only have to interop with myself in this case, so strictly adhering to the standard isn't as much a concern.  Thanks.
